snuba/datasets.py

```diff
@@ -0,0 +1,208 @@
+import re, six
+
+from snuba import clickhouse, schemas, state, util, processor
+from snuba.clickhouse import escape_col
+
+class DataSet(object):
+    """
+    A DataSet defines the complete set of data sources, schemas, and
+    transformations that are required to:
+        - Consume, transform, and insert data payloads from Kafka into Clickhouse.
+        - Define how Snuba API queries are transformed into SQL.
+    """
+    def __init__(self, *args, **kwargs):
+        self.SCHEMA = None
+        self.PROCESSOR = None
+        self.PROD = False
+
+    def condition_expr(self, condition, body):
+        """
+        If this dataset has a particular way of turning an individual
+        (column, operator, literal) condition tuple into SQL, return it here.
+        """
+        return None
+
+    def column_expr(self, column_name, body):
+        """
+        If this dataset has a particular way of turning a column name
+        into a SQL expression (eg. special column aliases that evaluate
+        to something else), return it here.
+        """
+        return None
```

Comment:
Is returning `None` for these ever valid for a dataset? Should we `NotImplementedError` to avoid footguns?

Reply:
Nah this is fine, `util.column_expr` calls out to this just in case there are custom column definitions for the dataset, but if this returns none, we use the common ones still (like the `QUOTED_LITERAL` stuff etc)... but yeah, as in my comment above, perhaps this base class is where the common stuff should be implemented instead of util?